File: src/main/java/org/eightsines/decompiled/A37.java
package org.eightsines.decompiled;

import java.util.ArrayList;
import java.util.List;

public final class A37 {
    private final void f(ArrayList l) {
        List $this$fastForeach$iv = (List) l;
        int i$iv = 0;

        for (int var1 = $this$fastForeach$iv.size(); i$iv < var1; ++i$iv) {
            int it = ((Number) $this$fastForeach$iv.get(i$iv)).intValue();
            System.out.println("a = " + it);
        }
    }
}

File: src/main/kotlin/org/eightsines/kotlinperfopt/A37.kt
package org.eightsines.kotlinperfopt

// https://habr.com/ru/company/oleg-bunin/blog/420143/

inline fun <reified T> List<T>.fastForeach(crossinline action: (T) -> Unit) {
    for (i in 0 until size) {
        action(this[i])
    }
}

class A37 {
    private fun f(l: ArrayList<Int>) {
        l.fastForeach {
            val a = it
            println("a = $a")
        }
    }
}
